docusaurus/docs/how-to-guides/data-source-plugins/config-test.md

```diff
@@ -0,0 +1,135 @@
+---
+id: data-source-configuration
+title: Configure data sources
+description: A guide to data source configuration.
+keywords:
+  - grafana
+  - data source plugin
+  - datasource
+  - data source plugin configuration
+---
+
+# Best practices for data source plugin configuration
+
+To deliver a consistent and user-friendly experience to Grafana users, we recommend that Grafana plugin developers follow uniform guidelines when building data source plugins. This guide outlines recommended best practices to help you achieve this goal.
+
+## Logos
+
+If your plugin requires a logo, follow these recommendations:
+
+- Ensure you obtain permission from the trademark holder.
+- Follow the brand guidelines, typically available in a company’s media kit or press kit.
+- Use logos with a transparent background, avoiding white backgrounds.
+- Logos should be in SVG format.
+
+**Example:** MongoDB  
+**Example:** Datadog
+
+## Description and required fields
+
+We recommend using the `DataSourceDescription` component from the [@grafana/experimental](https://github.com/grafana/grafana-experimental) repository. The plugin name and documentation link are required, but details about other fields is optional.
+
+For required fields, avoid manually typing `*`. Instead, pass the `required` prop to the `InlineField` component. This prop automatically adds a `*` to the label.
+
+**Example of DataSourceDescription component:**
+
+![Description](/img/data-source-config/config-data-source-description.png)
+
+**Example of required field:**
+
+![Connection](/img/data-source-config/config-connection.png)
+
+## Sections
+
+Divide the data source configuration page into multiple sections using the `ConfigSection` component.
+
+- The `section` description property is optional but recommended if the section contains more than one field.
+- For sections with optional or rarely used settings, it's best to keep them collapsed by default.
+- Use the `ConfigSubSection` component for organizing subsections within a section. Avoid separating subsections with horizontal lines.
+
+**Example of sections:**
+
+![Sections](/img/data-source-config/config-sections.png)
+
+## Authentication component
+
+Use the `Authentication` component for handling authentication on the configuration page. This component is already wrapped inside a `ConfigSection`, so additional wrapping is unnecessary.
+
+If your data source uses the deprecated `DataSourceHttpSettings` component for authentication, replace it with the `Authentication` component. For guidance on migrating from `DataSourceHttpSettings`, see the section below.
```

Comment:
Is `DataSourceHttpSettings` really deprecated? 

I can't find anywhere where is tagged as deprecated. it is not mention in the source code https://github.com/grafana/grafana/blob/main/packages/grafana-ui/src/components/DataSourceSettings/DataSourceHttpSettings.tsx or in the story page docs https://developers.grafana.com/ui/latest/index.html?path=/docs/data-source-datasourcehttpsettings--docs and furthermore we actually recommend this component as the easiest way to go to configure authentication

Reply:
Deprecated apparently isn't the right word. Additionally, what you are saying is that the advice given in this topic is just wrong, and we want people to use this component. Let me know what the advice is and I'll rewrite.

Reply:
`DataSourceHttpSettings` is still the recommended way to configure the datasource API endpoint and auth. There are some efforts in some internal Grafana teams to replace it with other components (the ones from experimental) but until that leaves the grafana-experimental package and we provide full documentation about it we can't recommend them over `DataSourceHttpSettings`

Reply:
I think the new component got moved to `@grafana/plugin-ui` last week and there is proper documentation on how to migrate to it from `DataSourceHttpSettings`.

https://github.com/grafana/plugin-ui/blob/main/src/components/ConfigEditor/Auth/README.md

But not sure it is officially the way to go yet.

Reply:
ok. i've added a link to the migration doc that you gave. i think this incorporates all the feedbak received so far, but i'm happy to leave this PR out for review for a few more days.

Reply:
Maybe depreciated isn't the right word, but the old DataSourceHttpSettings is not really recommended as it does have UX issues and historically the team was advising developers to switch.

Reply:
We should keep the call to use `DataSourceHttpSettings` until we properly document it in this document and update some examples to use the new components, should we actually want to do it.